lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -474,6 +469,94 @@ public static function get_user_data_from_wp_global_styles( $theme, $create_post
 		return $user_cpt;
 	}
 
+	// @codingStandardsIgnoreStart Squiz.Commenting.FunctionComment.ParamCommentFullStop
+	/**
+	 * Saves a new user variation into the database.
+	 *
+	 *
+	 * @param array $args Arguments. All are required.
+	 *     {
+	 *     @type string title Global styles variation name.
+	 *     @type string global_styles Global styles settings as a JSON string.
+	 *     @type string $stylesheet Slug of the theme associated with these global styles.
+	 * }
+	 *
+	 * @return int|WP_Error Post ID of the new variation or error if insertion failed.
+	 */
+	public static function add_user_global_styles_variation( $args ) {
+		$theme = wp_get_theme();
+
+		/*
+		 * Bail early if the theme does not support a theme.json.
+		 *
+		 * Since wp_theme_has_theme_json only supports the active
+		 * theme, the extra condition for whether $theme is the active theme is
+		 * present here.
+		 */
+		if ( $theme->get_stylesheet() === $args['stylesheet'] && ! wp_theme_has_theme_json() ) {
+			return new WP_Error( __( 'Theme does not have theme.json', 'gutenberg' ) );
+		}
+
+		$post_id = wp_insert_post(
+			array(
+				'post_content' => $args['global_styles'],
+				'post_status'  => 'publish',
+				'post_title'   => $args['title'],
+				'post_type'    => 'wp_global_styles',
+				'post_name'    => sprintf( 'wp-global-styles-%s', urlencode( $args['stylesheet'] ) ),
+				'tax_input'    => array(
+					'wp_theme' => array( $args['stylesheet'] ),
+				),
+			),
+			true
+		);
+
+		return $post_id;
+	}
+	// @codingStandardsIgnoreEnd Squiz.Commenting.FunctionComment.ParamCommentFullStop
+
+	/**
+	 * Make an association between post $id and post containing current user
+	 * global styles
+	 *
+	 * @since 6.2
+	 *
+	 * @param int|null $id ID of the associated post. Null to delete the asociation.
+	 * @return int|false Meta ID on success, false on failure.
+	 */
+	public static function associate_user_variation_with_global_styles_post( $id ) {
+		$current_gs_id = static::get_user_global_styles_post_id();
+
+		if ( $id === $current_gs_id ) {
+			return false;
+		}
+
+		$prev_id = get_post_meta( $current_gs_id, 'associated_user_variation', true );
```

Comment:
Why do we need the "user styles" to be linked to a "style variation" this way?

Reply:
It's used in JS later to keep the changes that are made to custom styles in sync with the saved and associated custom style. Then when the user clicks the "Save" button in Gutenberg, they have the option to untick the checkbox that also pushes the changes to the associated style variation.

Reply:
👋🏻 what is AS and what is CUGS?

Reply:
AS = Associated User Style Variation
CUGS = Current User Global Styles

I've explained this terminology in the initial PR "comment", it's bad, we've decided to switch terminology but I don't want to edit all comments now.

Reply:
> but I don't want to edit all comments now.

why not? if you're convinced that it's bad one commit will stop these questions repeating 😁

Reply:
I'm referring to my comments here in GitHub, if there are problems in code comments or code itself that's a different thing but I don't think that's the case, I didn't refer to CUGS / AS in code comments.

Reply:
I misunderstood 👍🏻 🙇🏻

Reply:
Should we replace the if and the get post meta call with `get_associated_user_variation_id` which I see defined later?